### PR TITLE
Format queue info

### PR DESCRIPTION
**Pull request by luisg5**
### Summary

Addresses #509 

### Details and comments

This PR formats the job queue information to be more user friendly. Also, it introduces a new dependency `arrow` to help with the time conversion.

### CODE REVIEW

File: qiskit/providers/ibmq/job/queueinfo.py [hunk 1]
```diff
@@ -56,5 +63,29 @@ def __init__(
...
+        queue_info = [
+            "Job {} queue information:".format(self.job_id),
+            "    queue position: {}".format(self.position),
+            "    status: {}".format(job_status.value),
+            "    estimated start time: {}".format(estimated_start_time),
+            "    estimated completion time: {}".format(estimated_completion_time),
+            "    hub priority: {}".format(self.hub_priority),
+            "    group priority: {}".format(self.group_priority),
+            "    project priority: {}".format(self.project_priority)
+        ]
```
Comment (jyu00) [inline qiskit/providers/ibmq/job/queueinfo.py]: Technically any of these fields can be `None`. Can you change it to print `unknown` if the field is `None`?

File: qiskit/providers/ibmq/job/queueinfo.py [hunk 2]
```diff
@@ -73,19 +72,40 @@ def format(self) -> str:
...
-            "    group priority: {}".format(self.group_priority),
-            "    project priority: {}".format(self.project_priority)
+            "Job {} queue information:".format(attributes_map['job_id']),
+            "    queue position: {}".format(attributes_map['position']),
+            "    status: {}".format(attributes_map['_status']),
+            "    estimated start time: {}".format(attributes_map['estimated_start_time']),
+            "    estimated completion time: {}".format(attributes_map['estimated_complete_time']),
+            "    hub priority: {}".format(attributes_map['hub_priority']),
+            "    group priority: {}".format(attributes_map['group_priority']),
+            "    project priority: {}".format(attributes_map['project_priority'])
```
Comment (jyu00) [inline qiskit/providers/ibmq/job/queueinfo.py]: I think it's better to avoid using strings as attribute names, since it is prone to typos not caught by the IDE and more easily missed when the attribute is renamed. You can instead write a small function that takes `foo` and returns `foo` if not `None` or "unknown".

**Comment by jyu00** [comment]
Can you also update `__repr__` (in this PR or another one) to say something like 
```
QueueInfo(_status='job is queued', estimated_complete_time=2020-01-16T15:56:19.079000+00:00 estimated_start_time=2020-01-16T15:56:19.079000+00:00, group_priority=1.0, hub_priority=1.0, position=5, project_priority=1.0)
```

**Comment by luisg5** [comment]
> QueueInfo(_status='job is queued', estimated_complete_time=2020-01-16T15:56:19.079000+00:00 estimated_start_time=2020-01-16T15:56:19.079000+00:00, group_priority=1.0, hub_priority=1.0, position=5, project_priority=1.0)

Yeah, I'll go ahead and address it in this PR!

File: qiskit/providers/ibmq/job/queueinfo.py [hunk 3]
```diff
@@ -56,5 +62,47 @@ def __init__(
...
+
+        return '\n'.join(queue_info)
+
+    def _get_value(self, value: Optional[Any], _default_value: str = 'unknown') -> Optional[Any]:
+        """Returns the value if it exists or a default.
+
+        Returns:
+            The value if it is not None, else a default value.
+        """
+        return value if value is not None else _default_value
```
Comment (jyu00) [inline qiskit/providers/ibmq/job/queueinfo.py]: nit: you can shorten this to `return value or _default_value`. And why is `_default_value` private?

File: qiskit/providers/ibmq/job/queueinfo.py [hunk 4]
```diff
@@ -56,5 +62,47 @@ def __init__(
...
+        self.job_id = job_id
 
         super().__init__(**kwargs)
+
+    def __repr__(self) -> str:
+        """"""
+        queue_info = ["{}={}".format(attr, repr(value))
+                      for attr, value in self.__dict__.items()]
+
+        return "{}({})".format(self.__class__.__name__, ', '.join(queue_info))
```
Comment (jyu00) [inline qiskit/providers/ibmq/job/queueinfo.py]: Can you add `<>` around the string to match Python standard?

File: qiskit/providers/ibmq/job/queueinfo.py (hunk 3 above)
Comment (luisg5) [reply qiskit/providers/ibmq/job/queueinfo.py]: Thanks for the tip @jyu00! I was just trying to convey that it's not really something to change, but I don't think it matters if it's changed by the user...

File: qiskit/providers/ibmq/job/queueinfo.py [hunk 5]
```diff
@@ -71,7 +71,7 @@ def __repr__(self) -> str:
         queue_info = ["{}={}".format(attr, repr(value))
                       for attr, value in self.__dict__.items()]
 
-        return "{}({})".format(self.__class__.__name__, ', '.join(queue_info))
+        return "<{}({})>".format(self.__class__.__name__, ', '.join(queue_info))
```
Comment (jyu00) [inline qiskit/providers/ibmq/job/queueinfo.py]: Can you actually format the datetime object to a more readable format? Like `'2020-01-17T23:29:47.684000+00:00'` (i.e. `.isoformat()`). If you just do a `str()` it looks like `datetime.datetime(2020, 1, 17, 23, 29, 47, 684000, tzinfo=datetime.timezone.utc)`, which is no different from the default representation. Same goes to status - it should be the value of the enum, which is more readable.

File: qiskit/providers/ibmq/job/queueinfo.py [hunk 6]
```diff
@@ -56,5 +62,47 @@ def __init__(
         self.hub_priority = hub_priority
         self.group_priority = group_priority
         self.project_priority = project_priority
+        self.job_id = job_id
 
         super().__init__(**kwargs)
+
+    def __repr__(self) -> str:
+        """"""
```
Comment (luisg5) [inline qiskit/providers/ibmq/job/queueinfo.py]: I almost missed this empty doctoring 😅

**Comment by luisg5** [comment]
As an update, `QueueInfo.__repr__` currently prints the following:

```python
<QueueInfo(_status='job is queued', estimated_complete_time='2020-01-17T22:59:25.907000+00:00', project_priority='1.0', job_id='5e22013e69f2d4001830bbbd', estimated_start_time='2020-01-17T22:58:54.507000+00:00', hub_priority='1.0', group_priority='1.0', position='75')>
```

And, when there are fields missing it will be similar to:

```python
<QueueInfo(_status='unknown', estimated_complete_time='2020-01-17T22:59:25.907000+00:00', project_priority='unknown', job_id='5e22013e69f2d4001830bbbd', estimated_start_time='unknown', hub_priority='1.0', group_priority='unknown', position='75')>
```

File: qiskit/providers/ibmq/job/queueinfo.py [hunk 7]
```diff
@@ -56,5 +62,56 @@ def __init__(
...
+        Returns:
+            a string representation of QueueInfo.
+        """
+        queue_info = []
+        for attr, value in self.__dict__.items():
+            if isinstance(value, datetime):
+                value = value.isoformat()
+            elif value in [status.value for status in ApiJobStatus]:
+                value = api_status_to_job_status(ApiJobStatus(value)).value
+            queue_info.append("{}='{}'".format(attr, self._get_value(value)))
```
Comment (jyu00) [inline qiskit/providers/ibmq/job/queueinfo.py]: Since the point of the custom `__repr__` is to be user friendly, can we make it more so by grouping related fields together in order? So instead of 
```
<QueueInfo(_status='job is queued', estimated_complete_time='2020-01-17T22:59:25.907000+00:00', project_priority='1.0', job_id='5e22013e69f2d4001830bbbd', estimated_start_time='2020-01-17T22:58:54.507000+00:00', hub_priority='1.0', group_priority='1.0', position='75')>
```
Can we have
```
<QueueInfo(job_id='5e22013e69f2d4001830bbbd', _status='job is queued', estimated_start_time='2020-01-17T22:58:54.507000+00:00', estimated_complete_time='2020-01-17T22:59:25.907000+00:00', position='75', hub_priority='1.0', group_priority='1.0', project_priority='1.0')>
```

Comment (luisg5) [reply qiskit/providers/ibmq/job/queueinfo.py]: Yeah, definitely! That'd be a nice feature. I'll go ahead and update the PR.

Comment (luisg5) [reply qiskit/providers/ibmq/job/queueinfo.py]: Yeah definitely! That would be a nice feature to have. I'll go ahead and update the PR!

File: qiskit/providers/ibmq/job/queueinfo.py [hunk 8]
```diff
@@ -72,13 +72,23 @@ def __repr__(self) -> str:
...
+
+        queue_info = [
+            "job_id: '{}'".format(self._get_value(self.job_id)),
+            "_status: '{}'".format(self._get_value(status)),
+            "estimated_start_time: '{}'".format(estimated_start_time),
+            "estimated_complete_time: '{}'".format(estimated_complete_time),
+            "position: '{}'".format(self._get_value(self.position)),
+            "hub_priority: '{}'".format(self._get_value(self.hub_priority)),
+            "group_priority: '{}'".format(self._get_value(self.group_priority)),
+            "project_priority: '{}'".format(self._get_value(self.project_priority))
```
Comment (jyu00) [inline qiskit/providers/ibmq/job/queueinfo.py]: Can you change the formatting to `foo=bar` instead of using a semicolon, to match the python convention?

File: qiskit/providers/ibmq/job/queueinfo.py [hunk 9]
```diff
@@ -56,5 +62,66 @@ def __init__(
...
+
+        queue_info = [
+            "job_id='{}'".format(self._get_value(self.job_id)),
+            "_status='{}'".format(self._get_value(status)),
+            "estimated_start_time='{}'".format(estimated_start_time),
+            "estimated_complete_time='{}'".format(estimated_complete_time),
+            "position='{}'".format(self._get_value(self.position)),
+            "hub_priority='{}'".format(self._get_value(self.hub_priority)),
+            "group_priority='{}'".format(self._get_value(self.group_priority)),
+            "project_priority='{}'".format(self._get_value(self.project_priority))
```
Comment (jyu00) [inline qiskit/providers/ibmq/job/queueinfo.py]: nit - you don't actually need to quote `position` and `priority`'s, since they're not strings.

**Comment by luisg5** [comment]
Thanks for all of the comments and guidance @jyu00!
